editor/store/effects.js

```diff
@@ -95,239 +79,17 @@ export function removeProvisionalBlock( action, store ) {
-			noticeMessage = __( 'Post reverted to draft.' );
-			shouldShowLink = false;
-		} else if ( ! isPublished && willPublish ) {
-			// If publishing or scheduling a post, show the corresponding
-			// publish message
-			noticeMessage = {
-				publish: __( 'Post published!' ),
-				private: __( 'Post published privately!' ),
-				future: __( 'Post scheduled!' ),
-			}[ post.status ];
-		} else {
-			// Generic fallback notice
-			noticeMessage = __( 'Post updated!' );
-		}
-
-		if ( noticeMessage ) {
-			dispatch( createSuccessNotice(
-				<p>
-					{ noticeMessage }
-					{ ' ' }
-					{ shouldShowLink && <a href={ post.link }>{ __( 'View post' ) }</a> }
-				</p>,
-				{ id: SAVE_POST_NOTICE_ID, spokenMessage: noticeMessage }
-			) );
-		}
-	},
-	REQUEST_POST_UPDATE_FAILURE( action, store ) {
-		const { post, edits, error } = action;
-
-		if ( error && 'rest_autosave_no_changes' === error.code ) {
-			// Autosave requested a new autosave, but there were no changes. This shouldn't
-			// result in an error notice for the user.
-			return;
-		}
-
-		const { dispatch } = store;
-
-		const publishStatus = [ 'publish', 'private', 'future' ];
-		const isPublished = publishStatus.indexOf( post.status ) !== -1;
-		// If the post was being published, we show the corresponding publish error message
-		// Unless we publish an "updating failed" message
-		const messages = {
-			publish: __( 'Publishing failed' ),
-			private: __( 'Publishing failed' ),
-			future: __( 'Scheduling failed' ),
-		};
-		const noticeMessage = ! isPublished && publishStatus.indexOf( edits.status ) !== -1 ?
-			messages[ edits.status ] :
-			__( 'Updating failed' );
-		dispatch( createErrorNotice( noticeMessage, { id: SAVE_POST_NOTICE_ID } ) );
+	REQUEST_POST_UPDATE: ( action, store ) => {
+		requestPostUpdate( action, store );
 	},
-	TRASH_POST( action, store ) {
-		const { dispatch, getState } = store;
-		const { postId } = action;
-		const basePath = wp.api.getPostTypeRoute( getCurrentPostType( getState() ) );
-		dispatch( removeNotice( TRASH_POST_NOTICE_ID ) );
-		apiFetch( { path: `/wp/v2/${ basePath }/${ postId }`, method: 'DELETE' } )
-			.then( () => {
-				const post = getCurrentPost( getState() );
-
-				// TODO: This should be an updatePost action (updating subsets of post properties),
-				// But right now editPost is tied with change detection.
-				dispatch( resetPost( { ...post, status: 'trash' } ) );
-			} )
-			.catch( ( error ) => dispatch( {
-				...action,
-				type: 'TRASH_POST_FAILURE',
-				error,
-			} ) );
-	},
-	TRASH_POST_FAILURE( action, store ) {
-		const message = action.error.message && action.error.code !== 'unknown_error' ? action.error.message : __( 'Trashing failed' );
-		store.dispatch( createErrorNotice( message, { id: TRASH_POST_NOTICE_ID } ) );
+	REQUEST_POST_UPDATE_SUCCESS: requestPostUpdateSuccess,
+	REQUEST_POST_UPDATE_FAILURE: requestPostUpdateFailure,
+	TRASH_POST: ( action, store ) => {
+		trashPost( action, store );
```

Comment:
I can't assign it directly because there's a warning if we return a promise from effect handlers